### PR TITLE
Add sidekiq middleware

### LINKED ISSUE #36: Sidekiq middleware
Add Sidekiq middleware that reports errors in failing jobs

### PR DESCRIPTION
**Pull request by brunvez**
Closes #36

### CODE REVIEW

File: lib/exception_hunter/middleware/request_hunter.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
+module ExceptionHunter
+  module Middleware
+    class RequestHunter
```
Comment (t-romani) [inline lib/exception_hunter/middleware/request_hunter.rb:3]: why this change?

File: lib/exception_hunter/middleware/sidekiq_hunter.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      ARGS_TRACKED_DATA = %w[
+        job_class
+        job_id
+        arguments
+        enqueued_at
+      ].freeze
+
+      def call(_worker, context, _queue)
+        yield
+      rescue Exception => exception # rubocop:disable Lint/RescueException
```
Comment (t-romani) [inline lib/exception_hunter/middleware/sidekiq_hunter.rb:21]: 2old school

File: spec/helpers/exception_hunter/errors_helper_spec.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+          something: 1,
+          nested: [
+            'abc',
+            { email: 'example@email.com' }
+          ]
+        }
+      end
+
+      it 'returns a json string with correct indentation' do
+        # rubocop:disable Layout/LineLength
```
Comment (t-romani) [inline spec/helpers/exception_hunter/errors_helper_spec.rb:17]: why not editing the yml?

File: spec/middleware/exception_hunter/sidekiq_hunter_spec.rb [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+          let(:worker) { -> { raise 'Some error' } }
+
+          it 're-raises the exception' do
+            expect { subject }.to raise_error(RuntimeError, 'Some error')
+          end
+
+          it 'tracks the error' do
+            expect {
+              begin
+                       subject
```
Comment (t-romani) [inline spec/middleware/exception_hunter/sidekiq_hunter_spec.rb:38]: weird indentation

File: spec/middleware/exception_hunter/sidekiq_hunter_spec.rb [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+              {
+                'queue' => 'default',
+                'retry_count' => '3'
+              }
+            end
+
+            it 'tracks the error' do
+              expect {
+                begin
+                         subject
```
Comment (t-romani) [inline spec/middleware/exception_hunter/sidekiq_hunter_spec.rb:75]: weird indentation

File: spec/middleware/exception_hunter/sidekiq_hunter_spec.rb [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+              {
+                'queue' => 'default',
+                'retry_count' => '2'
+              }
+            end
+
+            it 'does not create an error' do
+              expect {
+                begin
+                         subject
```
Comment (t-romani) [inline spec/middleware/exception_hunter/sidekiq_hunter_spec.rb:94]: 🤔

File: spec/middleware/exception_hunter/sidekiq_hunter_spec.rb [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+          end
+        end
+
+        context 'when the worker does not raise an exception' do
+          let(:worker) { -> {} }
+
+          it 'does not create an error' do
+            expect {
+              begin
+                       subject
```
Comment (t-romani) [inline spec/middleware/exception_hunter/sidekiq_hunter_spec.rb:109]: 🤷‍♀️

File: lib/exception_hunter/middleware/sidekiq_hunter.rb (hunk 2 above)
Comment (brunvez) [reply lib/exception_hunter/middleware/sidekiq_hunter.rb:21]: 👴 I like to disable linters close to the code so people can fix them if they see something I did not. Plus moving code around breaks YML rules

File: lib/exception_hunter/middleware/request_hunter.rb (hunk 1 above)
Comment (brunvez) [reply lib/exception_hunter/middleware/request_hunter.rb:3]: Small refactor, we'll have many more middlewares in the future I thought it'd be nice to have a place for them
